oauthlib/oauth2/rfc6749/parameters.py

```diff
@@ -118,10 +137,21 @@ def prepare_token_request(grant_type, body='', **kwargs):
     if 'scope' in kwargs:
         kwargs['scope'] = list_to_scope(kwargs['scope'])
 
+    # pull the `client_id` out of the kwargs. 
+    client_id = kwargs.pop('client_id', None)
+    if include_client_id:
+        if client_id is not None:
```

Comment:
It seems that `client_id` isn't used in this function except in this line... I'm not sure if I understand why we'd need `include_client_id` in this function's signature in that case.

Reply:
It feels a bit fiddly with the `None`, `False`, `True` options and I'd be inclined to put the logic in the individual client classes, but am happy with this too.